### ISSUE TITLE
KeePassXC Won't Copy Password When "Lock Database After Minimizing Window" Is Checked.

### ISSUE THREAD

**Issue by Zackery** [keepassxreboot/keepassxc#1090]
## Current/Expected Behavior
Hello, after enabling the option "Lock Database After Minimizing Window", whenever I copy a password, KeePass closes normally (because I have "Minimize when copying to clipboard" checked), however if I try to paste my password, nothing is in my clipboard, when it should be.

## Steps to Reproduce (for bugs)
1. Enable "Lock databases after minimizing window".
2. Copy password, KeePass should minimize if the setting is enabled to where if you copy a password, the program minimizes.

Picture of settings. [[1](https://i.zelakto.tv/9ym0mno.png)] [[2](https://i.zelakto.tv/e4rXSss.png)]

## Context
Very inconvenient, the only way I can copy passwords is if I physically edit the entry, or disable the option to where the database locks once the program is minimized.

## Debug Info
KeePassXC - Version 2.2.1
Revision: 2bce9c8add07226e9a05e9e0fd0e5e66b236d5b6

Libraries:
- Qt 5.9.1
- libgcrypt 1.8.1

Operating system: Windows 10 (10.0)
CPU architecture: x86_64
Kernel: winnt 10.0.16299

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

**Comment by phoerious** [comment]
Hard to say if this is really a bug or expected behavior.

**Comment by TheZ3ro** [comment]
Seems expected/correct behavior to me

**Comment by r0ckarong** [comment]
How is this expected behavior if the function of "copy to clipboard" is not performed?

The expected sequence would be (with the options active here):

1. Request copy to clipboard
2. Password is copied to clipboard
3. Application is minimized
4. Database is locked

You're saying by activating this combination of options, Step 2 is skipped and this is "correct" behavior. 

When 1 is "Copy this" and 2 is "Is copied", then you can't have any "correct sequence" that starts with "Copy this" and not ever perform a copy no matter what the following steps are.

What the user is describing here is the sequence actually works like this:

1. Request copy to clipboard
2. Application is minimized
3. Database is locked

**Comment by phoerious** [comment]
It is copied, but closing the database clears the clipboard.

**Comment by TheZ3ro** [comment]
If the database is locked there is no reason to keep data in the clipboard.

If we change the behavior in this case we need to keep in mind every other case (I personally want to clear clipboard if my database is locked)

**Comment by droidmonkey** [comment]
A modified behavior when these options are both enabled could be:

1. Copy to clipboard
2. Minimize to tray
3. Wait for clipboard timeout or 30 seconds whichever is less
4. Lock the database

**Comment by jeena** [comment]
KeePassX does what @droidmonkey describes.
Why would I copy something to the clipboard which I can't paste because it get's deleted within the same milisecond?

**Comment by TheZ3ro** [comment]
@jeena And why would KeePassXC leave the DB unlocked when I have explicitly enabled "Lock db after minimizing windows" ?

IMHO this should be a Security option and by default KeePassXC should always wipe the clipboard when a DB is locked. 
I don't think KeePassX leave data in the clipboard after a DB has been locked, and if it does it's a security problem.

**Comment by leonardoporpora** [comment]
In keepass if you use Ctrl + C for copying the password it will disappear from clipboard in 15 s if I remember well (15 or 10)

**Comment by jeena** [comment]
@TheZ3ro Because otherwise this functionality is unusable, what's the point of having this function if you can't do anything with it, or am I missing something?

The only thing I want to do is to be able to paste the password into the other app. And I'd like the db to be automatically locked so I don't need to do it manually every time.

How do you use KeePassXC, do you lock the database manually after you pasted the password?

**Comment by droidmonkey** [comment]
I personally think the lock on minimize feature is silly and results in you having to unlock the database way too much.

**Comment by TheZ3ro** [comment]
I personally use autotype most of the time. 
The few times I copy to clipboard I don't lock the DB since I have autolock after 60 sec (and clear-clipboard after 20 sec)

The functionalities are not usable only when used together. 
To me it make no sense to copy a password, minimize to tray (and thus locking the DB since the option is enabled) and assuming the password is still in the clipboard. 

When a db is locked, no data should be available (clipboard included)

**Comment by jeena** [comment]
It was quite some time ago so I think it was both, an older version of KeePassX and I used the perform autotype function.

**Comment by phoerious** [comment]
Is this still an issue?

**Comment by droidmonkey** [comment]
Yes it is

**Comment by Cornelius-Riemenschneider** [comment]
This issue is especially annoying in my setup:
1. KeepassXC database is locked
2. I request auto-type via shortcut
3. The website I'm on has only a field for password (not username+pw), e.g. paypal.com
4. I right-click on the KeepassXC popup to only copy the password entry
5. The DB is auto-locked because auto-type is done, and the clipboard is cleared

Thus, the auto-type copy-to-clipboard feature is totally broken in conjunction with the close-db-after-auto-type-if-db-was-closed-before security feature.

**Comment by Mardiie** [comment]
I don't want to be too negative, especially since the team that works on this project has already gotten me banned once after I voiced concerns about not having windows hello support on the project (lo and behold, it was added) and feedback was not welcomed. 
These options are exactly the same as in the original KeePass and once again it seems like the people working on KeePassXC dropped the ball completely. As people have mentioned in this post the behavior currently in the app is totally counterintuitive and just plain wrong following normal logic. However one can play a nice game of ping pong with KeePassXC by copying a password, having auto-minimize on, and then having no password to paste. Rinse and repeat. Fun little game!
